temoa/extensions/discrete_capacity/components/discrete_capacity.py

```
@@ -0,0 +1,90 @@
+
+
+def limit_discrete_new_capacity_indices(
+    model: DiscreteCapacityModel,
+) -> set[tuple[Region, Technology, Vintage]]:
+    indices = {
+        (r, t, v)
+        for r, t in model.limit_discrete_new_capacity.sparse_keys()
+        for v in model.vintage_optimize
+        if capacity.gather_group_built_processes(model, r, t, v)
+    }
+    return indices
```

Ruff (0.15.21)

[warning] Unnecessary assignment to `indices` before `return` statement

Remove unnecessary assignment

(RET504)